doc/rst/technotes/gpu.rst

```diff
@@ -136,39 +139,125 @@ GPU-Related Environment Variables
+code for and interacts with GPUs. These variables include:
+
+* ``CHPL_GPU`` --- may be set to ``nvidia``, ``amd', or ``cpu``. If unset, as
+  part of its build process, Chapel will attempt to automatically determine
+  what type of GPU you're trying to target. Changing this variable requires
+  rebuilding Chapel. For more information, see the `Vendor Portability`_
+  section.
+* ``CHPL_GPU_ARCH`` --- specifies GPU architecture to generate kernel code for.
+  If unset and targeting NVIDIA GPUs, will default to 'sm_60'. This may also
+  be set by passing the ``chpl`` compiler ``--gpu-arch=<architecture>``. For
+  more information, see the `Vendor Portability`_ section.
+* ``CHPL_CUDA_PATH`` --- specifies path to CUDA toolkit.  If unset, Chapel tries
+  to automatically determine this path based on the location of ``nvcc``. This
+  variable is unused if not targeting NVIDIA GPUs. For more information, see
+  the `Vendor Portability`_ section.
+* ``CHPL_ROCM_PATH`` --- specifies path to ROCm library. If unset, Chapel tries
+  to automatically determine this path based on the location of ``hipcc``. This
+  variable is unused if not targeting AMD GPUs. For more information, see the
+  `Vendor Portability`_ section.
+* ``CHPL_RT_NUM_GPU_PER_LOCALE`` --- if using ``CHPL_GPU=cpu``, sets how many
+  GPU sublocales to have per locale. For more information, see the `CPU as
+  Device mode`_ section.
+* ``CHPL_GPU_MEM_STRATEGY`` --- dictates how to allocate data when on a GPU
+  locale.  May be set to ``unified_memory`` or ``array_on_device``. If unset,
+  defaults to ``unified_memory``. Changing this variable requires rebuilding
+  Chapel. For more information, see the `Memory Strategies`_ section.
+* ``CHPL_GPU_BLOCK_SIZE`` --- specifies default block size when launching
+  kernels. If unset, defaults to 512. This variable may also be set by passing
+  the ``chpl`` compiler ``--gpu-block-size=<block_size>``. It can also be
+  overwritten on a per-kernel basis by using the :proc:`~GPU.setBlockSize`
+  function.
+* ``CHPL_GPU_SPECIALIZATION`` --- if set, outlines bodies of 'on' statements
+  and clones all functions reachable from that block. The 'on' statement is
+  rewritten to call the cloned version of the outlined function when on a  GPU
+  locale. This mode increases overall code size but allows the compiler to
+  assume that a given function will execute on the GPU locale and optimize
+  accordingly. This may also be set by passing the ``chpl`` compiler the
+  ``--gpu-specialization`` flag. This is an experimental mode subject to change
+  in the future.
+* ``CHPL_GPU_NO_CPU_MODE_WARNING`` - this variable is relevant when using the
+  `CPU as Device mode`_ and if set causes it so that uses of
+  :proc:`~GPU.assertOnGpu` while in this mode will not generate a warning at
+  execution time. Alternatively, this can be done by passing
+  ``--gpuNoCpuModeWarning`` to your application. For more information, see the
+  `CPU as Device mode`_ section.
```

Comment:
```suggestion
* ``CHPL_GPU_NO_CPU_MODE_WARNING`` - this variable is relevant when using the
  `CPU as Device mode`_ and if set causes uses of
  :proc:`~GPU.assertOnGpu` to not generate a warning at
  execution time. Alternatively, this behavior can be enabled by passing
  ``--gpuNoCpuModeWarning`` to your application. For more information, see the
  `CPU as Device mode`_ section.
```
Forgive my line breaking.

Reply:
Huh, I wasn't aware GH would let you post suggestions like that with a "Commit suggestion" button to just apply it.

I'm not going to hit it because I'm updating things in my editor (and I'm not sure if the DCO line for my commit message would get through), but that's pretty cool.